### ISSUE TITLE
Solution for PVP

### ISSUE THREAD

**Issue by Battlepixel** [tsunamayo/Starship-EVO#1978]
PVP has two main **problems** with small ships:

1.) It is possible to build small ships with brick reactors and brick thrusters that accelerate to 400m / s or faster in 3s. These are too hard to hit and therefore overpowered. The problem here is not the top speed but the acceleration.

2.) It is also possible with normal blocks to build a space fighter that is only one cuboid filled with system blocks and that does not overheat -> speed, armament & shields scale too much with the cube shape of ships. The more cube-shaped a ship the stronger (this effect should not be as significant as it is currently)

**Solution** with minimal negative impact on other gameplay aspects
- Scale the output of brick system blocks to 0.75 of the output of large system blocks (if you are currently comparing thruster, the factor is 6.4 ^^)
- fix #1886 
- then increase the heat output of all system blocks by a factor of 5-8. Now the player with the better skills wins the PVP and not the one with the most cube-shaped ship^^) 

if you make these adjustments the fastest possible accelerations will be 300m / s in 20s, and according to the discussion with FD this is little and poor ^^ and therefore little enough. I think higher accelerations and top speeds are also possible, but this proposal is a compromise of different views.

**Comment by Battlepixel** [comment]
If you decide to make adjustments to the heatsystem, please keep also the previous build version online for a while so that players can customize their ships.
To fix the agility issue of big ships, I think a nice solution would be to adjust the function that calculates the acceleration from mass and thrust. ( maybe it should be not linear, so bigger things get slower and more sluggish than small things)

**Comment by tsunamayo** [comment]
> * Scale the output of brick system blocks to 0.75 of the output of large system blocks (if you are currently comparing thruster, the factor is 6.4 ^^)
> * then increase the heat output of all system blocks by a factor of 5-8. Now the player with the better skills wins the PVP and not the one with the most cube-shaped ship^^)


Yes I agree with the acceleration, I am on it. But I dont understand your first suggestion. Also I dont see how the last one could make it better, and it will destroy every single build for no reason.
Also not sure to understand why a cube is more heat efficient for a given volume:
A cube of 3x3x3 = 27 m3 has a 54m2 surface (6 x 3x3)
A long cube of 1x1x27 = 27m3 has a 110m2 surface (4 x 1x27 + 2x1)
Again before jumping too fast on making make change balance value you have to convince me first ;)
Cheers, and thanks for testing PVP. As soon as I finish the tutos I am doing that combat update.

**Comment by Battlepixel** [comment]
> I dont understand your first suggestion.

the bricksystemblock-output per volume is much larger then the output of the blocksystem. If you are using only block systemblocks you can only
![Starship EVO_2020_04_30_16_02_39_956](https://user-images.githubusercontent.com/43222403/81237965-28acc600-9001-11ea-9b52-1d8c34888609.jpg)
![Starship EVO_2020_04_30_16_03_05_533](https://user-images.githubusercontent.com/43222403/81237973-2d717a00-9001-11ea-8246-c03ed05d61dd.jpg)


> Also not sure to understand why a cube is more heat efficient for a given volume:
> A cube of 3x3x3 = 27 m3 has a 54m2 surface (6 x 3x3)
> A long cube of 1x1x27 = 27m3 has a 110m2 surface (4 x 1x27 + 2x1)

It isnt but at the moment there is no reason to build heatefficient, because the heatoutput is so small that you can fill qubic ships. 
The heat output is so small that it hardly plays a role in the construction. It is possible to fill the molds with minimal surface (cube ball) only with system blocks without overheating them. For PVP this is of course much better than building something nice that looks less cube-shaped and more like a space fighters because most cool designs have less volume with the same length, width and height. So the critic of FD was, that everyone who wants to win pvp, will build something full of blocks (because its possible) and in order to be able to place as many shields and drive units as possible, a cube-shaped ship is the best choice if it should fit through a 20 * 20m hangar doo. So this will lead to a Universe where ships that can win pvps will allway look something like this 
![image](https://user-images.githubusercontent.com/43222403/81239112-2ef07180-9004-11ea-886d-22eaabcc093f.png)

**Comment by Battlepixel** [comment]
to clarify i'm the last one who wants any designs to stop working. I am in favor of the fact that as many designs as possible are competitive in pvp, of course also the cube-shaped ones who find them beautiful. I am only against the fact that you are forced to adhere to a strictly defined form in order to build a pvpfähiges ship. Of course, because I don't want my fighter planes to be cubes, but that's a matter of taste ^^

**Comment by Battlepixel** [comment]
What I entered in the googel translator was actually very logically structured and understandable. What the Googel translator spent was obviously not.^^ So if something is still unclear, this is not a mistake in the idea or the reasoning, but only in the communication. Just let me know, I'll try again ;)

**Comment by tsunamayo** [comment]
Yes this is why I created the Heat mechanic on the first place, to have people use block to create more interesting ship.
What I understand is I should decrease the surface heat bonus, and thats it. But the smaller ship I check where already having a high heat %, (and were not just cube), so I dont think there is room for that.
Please send me ship, example of such cube, and small ship that you think are inefficient compared to them...

**Comment by FlyingDebris** [comment]
To provide what I got out of our conversation, my point was that a couple of minor config changes should be made, instead of penalizing anyone that uses more than an exceedingly small portion of their ship's mass for systems. As a matter of fact, I'd say the heat mechanic is actually pretty decently balanced, and should be lower priority compared to thrust configs right now.

And no, my point was not that everyone would build borg cubes. As a matter of fact, my experience has shown that the PVP effectiveness of a ship has little to do with its shape except in edge cases,  and that it's more an issue of odd balancing. The point I was trying to make, when explaining how PVP players build ships, was that we don't leave the majority of our ships unfilled, or at the very least without purpose. I've attached a "meta" ship of mine, for reference. It's rather crudely built, but it gets the point across as to what you'd expect out of a basic PVP ship.

[TTS Draco CRV-501.zip](https://github.com/tsunamayo/Starship-EVO/files/4590346/TTS.Draco.CRV-501.zip)

**Comment by Battlepixel** [comment]
Oh yes we had some comunication probs^^.
Where we broke up was the point that I showed this shipexample that reach only 220m/s in 20s, depite being fullfill of systemblocks
![Starship EVO_2020_05_05_17_55_02_612](https://user-images.githubusercontent.com/43222403/81270620-4dc62680-904b-11ea-864b-c9ad49dfc888.jpg)
I think that shows that the block system (without brick reactors) is balanced, since we both thought that this ship was already accelerating very badly or in not my words poor and not made to win^^, but it is not possible to place any system block more here without thrroing out the gear or give up the design.
In this point we disagree, because FD was says he could build a fighter full of systemblocks(no bricks) and it would also be significantly to fast.

So the prob I have with the solution to decrease speed in generall is that already slow ships without bricks) get even slower and designs like this will never be able to win a pvp because there shape offers not enough space for thruster, power and shiedls:
![image](https://user-images.githubusercontent.com/43222403/81271950-122c5c00-904d-11ea-8d78-fba42e00fbde.png)
![Starship EVO_2020_05_02_16_17_38_676](https://user-images.githubusercontent.com/43222403/81272042-35570b80-904d-11ea-8168-cb6fc89c47aa.jpg)
![Starship EVO_2020_04_30_16_09_49_406](https://user-images.githubusercontent.com/43222403/81272067-40aa3700-904d-11ea-8e9a-be96e2798997.jpg)

**Comment by Battlepixel** [comment]
there are also easy-to-integrate alternatives that also solve all problems, but are less elegant because they make the heat system superfluous

**Comment by Arschpollo13** [comment]
[ "the acceleration is just 1.3 times higher than what you described as very poor(so no problem) and with coaxial guns it is clear that you got problems with hitting targets
if, contrary to what you said, you cannot upload anything more extremely from blocks, then the block balance is ok and does not have to be throttled(only bricks). So there is no need to adjust the heat output because a design-depending accelerationfactor of max. 1.5-2.0 is absolutely ok" quote Battlepixel ]

Then everything is fine, only the brick output must be redused.

**Comment by FlyingDebris** [comment]
I've got a ship illustrating my point a little more clearly than my first boat

unfortunately I blew it up in testing and can't teleport out of the void

will post when I can fix the guns lol

**Comment by FlyingDebris** [comment]
alright here's my shitbucket

it's alright I guess. tsuna if you want any explanations for why I did certain things let me know and I'll do my best to explain

cheers!

[Papa Giornos Pizza Factory.zip](https://github.com/tsunamayo/Starship-EVO/files/4602818/Papa.Giornos.Pizza.Factory.zip)

also something to note, saying it takes 20s to get to max speed is kinda disingenuous given that it takes 5 seconds to get to 3/4 of max speed. just food for thought though

**Comment by Battlepixel** [comment]
> also something to note, saying it takes 20s to get to max speed is kinda disingenuous given that it takes 5 seconds to get to 3/4 of max speed. just food for thought though
of course, it s only to compare more precise, the function is determinded if you know one other point than (0s, 0m/s²)

- your blueprint is accelerating faster than your last
- but its 67m long, so no prob to hit with movably mounted guns, its more like a corvett and no fighter
- it has coaxial guns so it will be difficult to hit something
- it has 5 to 8 times the mass of a fighter so if tsuna adjust the inertia-calc to make big things slower, it will loose acceleration
- your blueprint **uses only brickreactor**, there are OP you did say yourself (Poweroutput will not be enough if bricks get balanced, you will need more mass to produce enough power
-> So in my opinion this blueprint shows no problem that is an argument for putting the output under the current **block**-system-output/volume (not brick output this is OP I agree)
![Starship EVO_2020_05_09_10_05_47_237](https://user-images.githubusercontent.com/43222403/81468617-f48ffb80-91e0-11ea-9acc-32f1b623aa04.jpg)


if you disagree with me that only brickoutput/volume has to be reduced to blockoutput/volume but not even more, (in relation to the thrust output, the factor by which is reducedis 6.4!!)
do you have an alternative suggestion? 



> upload it together with a proposal that meets the following two criteria, or make it clear which criterion you contradict
> 
> **An alternative proposal would have to solve two problems:**
> 
> * ship strength for a given length-width-height should scale much less with volume than it currently does (so max design creativity is also possible in pvp)
> * It shouldn't force any design to a have a slower acceleration than 200m / s in 20s -> FDs words "this is very poor"^^ ->so its okey (if the proposal led to it, it would indirectly ban such designs)

**Comment by Sullos** [comment]
I believe the best solution is to simply solver the shielding balance issues, then to actually let combat play out between people. I am hearing all too often "That's OP! It needs to be nerfed" when the people involved have done such testing t the extent of building a big weapon, pointing at a stationary target and shooting it till it blows up. Trust me, I've done plenty of the same. However once you actually have moving targets with good AI, and other actual people, combat becomes very different. 
I suggest this topic be left to the side till we actually know there's a problem. I understand other games had problems, but I don't believe we're there yet.

**Comment by FlyingDebris** [comment]
> > also something to note, saying it takes 20s to get to max speed is kinda disingenuous given that it takes 5 seconds to get to 3/4 of max speed. just food for thought though
> 
> of course, it s only to compare more precise, the function is determinded if you know one other point than (0s, 0m/s)
> 
> * your blueprint is accelerating faster than your last
> * but its 67m long, so no prob to hit with movably mounted guns, its more like a corvett and no fighter
> * it has coaxial guns so it will be difficult to hit something
> * it has 5 to 8 times the mass of a fighter so if tsuna adjust the inertia-calc to make big things slower, it will loose acceleration
> * your blueprint **uses only brickreactor**, there are OP you did say yourself (Poweroutput will not be enough if bricks get balanced, you will need more mass to produce enough power
>   -> So in my opinion this blueprint shows no problem that is an argument for putting the output under the current **block**-system-output/volume (not brick output this is OP I agree)
>   ![Starship EVO_2020_05_09_10_05_47_237](https://user-images.githubusercontent.com/43222403/81468617-f48ffb80-91e0-11ea-9acc-32f1b623aa04.jpg)
> 
> if you disagree with me that only brickoutput/volume has to be reduced to blockoutput/volume but not even more, (in relation to the thrust output, the factor by which it would be reduced is already 6.4!!)
> do you have an alternative suggestion?
> 
> > upload it together with a proposal that meets the following two criteria, or make it clear which criterion you contradict
> > **An alternative proposal would have to solve two problems:**
> > 
> > * ship strength for a given length-width-height should scale much less with volume than it currently does (so max design creativity is also possible in pvp)
> > * It shouldn't force any design to a have a slower acceleration than 200m / s in 20s -> FDs words "this is very poor"^^ ->so its okey (if the proposal led to it, it would indirectly ban such designs)

I'm not really sure where to begin with this.

First off, your method of calculation only works well if acceleration is linear. It is not.

Second, good luck doing any meaningful damage to a ship like that, especially if it's more spread out than I did here, which was mostly because I wasn't trying to make a full on spaghetti ship. 

Third, I chose coaxial guns because turrets are currently broken. I'm well aware turrets are better than player control, believe me.

Fourth, I chose brick reactors specifically because they're better per volume.

The rest of your post borders on incomprehensible, and I can't figure out what you're suggesting. I will assume this is a translation issue, as you specified that your arguments are "logically structured and understandable. ^^"

What I am suggesting, in both long and short, is that acceleration become either linear or a cube root function. 

That's all. 

Issues with block power, brick power, what have you can be sorted out later, and have effectively nothing to do with the above. The games you've been basing your ideas off (Freelancer, 4X) are notorious for being unbalanced, (rightfully so) and I strongly suggest you look elsewhere for your gameplay ideas.

**Comment by Battlepixel** [comment]
> First off, your method of calculation only works well if acceleration is linear. It is not.

of cause it isnt linear^^ but you are wrong - If you have the function that you get by testing, one point is enough to know every other point on the function

> Second, good luck doing any meaningful damage to a ship like that, especially if it's more spread out than I did here, which was mostly because I wasn't trying to make a full on spaghetti ship.
+ try to build such shields and acceleration without brickreactor so...
> Issues with block power, brick power, what have you can be sorted out later,

 wrong! Every reactor I dont need, gives me more space for thruster und more speed(acceleration)
->look at this picture bro and you will understand
![Starship EVO_2020_05_05_17_55_02_612](https://user-images.githubusercontent.com/43222403/81472606-80ae1d00-91f9-11ea-884b-f33046108fe9.jpg)



> acceleration become either linear

x) As a result, low volume designs like for example: x-Wing, Tie-Fighters, F-302, Freelancer-fighter, Starciticen-fighter, and many more will accelerate even less than they already do. (they are accelerating really slow)
Since your suggestion certainly also affects the boost mode, there is another negative aspect that you can read more about here #1972 


> linear or a cube root function.
a cube root function has a strong negative effect on logic. 
if you don't believe me i will show you how much logic blocks you need at least to build an indestructible (by fast clicking a button or holding it down) circuit to change the order of two tasks alternately. For example, you usually need to work with landing flaps and gear.

But in my opinon a mass-root function could do together with balancing bricks. Thos was the point you disagree and this is the only reason for this suggestion. So if you agree now, we are ready, but if you dont, please make a other suggestion to slove point x).

**Comment by Battlepixel** [comment]
If your goal is also to get ahead faster and save time for everyone: 
- Please try not to ignore the counter arguments already mentioned to save time for everyone.
- Please no longer make claims that you have not checked or cannot check at all.

**Comment by Battlepixel** [comment]
> I believe the best solution is to simply solve the shielding balance issues, then to actually let combat play out between people. I am hearing all too often "That's OP! It needs to be nerfed" when the people involved have done such testing to the extent of building a big weapon, pointing at a stationary target, and shooting it till it blows up. Trust me, I've done plenty of the same. However once you actually have moving targets with good AI, and other actual people, combat becomes very different.
> 
> I suggest this topic be left to the side till we actually know there's a problem. I understand other games had problems, but I don't believe we're there yet.

Like I wrote, for me everything is fine if nobody suggests to make slow designs (F302,XWing...) even less slower.^^ Thats my point. 

Every other sentence and picture is to make clear that there are many ways around this.

**Comment by FlyingDebris** [comment]
> > First off, your method of calculation only works well if acceleration is linear. It is not.
> 
> * of cause it isnt linear^^ but you are wrong - If you have the function that you get by testing, (but you have not the parameters like thrust and mass because they are design-depending), one point is enough to know every other point on the function(to get the parameters or at least the quotient of mass and thrust)
> 
> > Second, good luck doing any meaningful damage to a ship like that, especially if it's more spread out than I did here, which was mostly because I wasn't trying to make a full on spaghetti ship.
> 
> * try to build such shields and acceleration without brickreactor so...
> 
> > Issues with block power, brick power, what have you can be sorted out later,
> 
> wrong! Every reactor I dont need, gives me more space for thruster und more speed^^(acceleration,)
> ->look at this picture bro and you will understand
> ![Starship EVO_2020_05_05_17_55_02_612](https://user-images.githubusercontent.com/43222403/81472606-80ae1d00-91f9-11ea-884b-f33046108fe9.jpg)
> 
> > acceleration become either linear
> 
> x) As a result, low volume designs like for example: x-Wing, Tie-Fighters, F-302, Freelancer-fighters, Starciticen-fighters, similar and many more will accelerate even less than they already do. (they are accelerating really slow)
> Since your suggestion certainly also affects the boost mode, there is another negative aspect that you can read more about here #1972 (for none-boost-mode it would be fine but that doesn't help you with pvp)
> 
> > linear or a cube root function.
> 
> a cube root function has a strong negative effect on logic.
> if you don't believe me i will show you how much logic blocks you need at least to build an indestructible (by fast clicking a button or holding it down) circuit to change the order of two tasks alternately. For example, you usually need to work with landing flaps and gear.
> 
> But in my opinon a mass-root function could do together with balancing bricks. Thos was the point you disagree and this is the only reason for this suggestion. So if you agree now, we are ready ;), but if you dont, please make a other suggestion to slove point x).

1. BASIC OBSERVATION of your speedometer ingame shows that I am right.

2. Already have. Look up "Fair and Balanced" in the discord to see.

3. Don't try to act smug when your english skills aren't that great. "Sorted out later" has nothing to do with what you just posted.

4. Get better at building ships. No amount of balance will change that someone else is going to properly fill their ship with systems, and you didn't. 

5. This has literally nothing at all to do with logic. I don't even know where you got this idea from.

6. As for your other two comments, they were completely unnecessary and repetitive.


Look, I've been trying to have constructive debates here, but your cyclical arguments and unwarranted smug attitude are getting us nowhere. I don't mean to be rude, but please just leave balance to the people on the right side of the dunning-kruger curve.

**Comment by Battlepixel** [comment]
> 6\. As for your other two comments, they were completely unnecessary and repetitive.

yeah, it would have been really easier if you had responsed to/gone into my arguments and show that you understood them . 
I have repeated it soon 10 times and you still have not taken a position on it, although I have asked you for it several times so I am really annoyed and dub it with what is called smug I think^^.

> 5\. This has literally nothing at all to do with logic. I don't even know where you got this idea from.

I got this Idea from testing, like I wrote. if 7-10% (only to have a not faulty landing gear with flaps) of the blocks of your ship are logic, it would make a significant difference in the blocknumber you suggested as base for acceleration-calc. But you are right here, this effect may be small and is therefore not a counter argument worth mentioning



> 4\. Get better at building ships. No amount of balance will change that someone else is going to properly fill their ship with systems, and you didn't.

would you just let yourself read what you just quoted in your last post


> 1. BASIC OBSERVATION of your speedometer ingame shows that I am right.

would you just let yourself read what you just quoted in your last post



> 2\. Already have. Look up "Fair and Balanced" in the discord to see.

please post it here if it isnt the first rectangular one


our discussion here is far too emotional

In the best case, tsuna simply reads through the course, finds the 5 arguments that we have distributed / or repeated on 100posts^^, and simply downloads the ships to form his own opinion.

**Comment by Incapelli** [comment]
Bruh. Can you stop? You're wasting everyone's time. Especially Tsuna's.

**Comment by Garrett-C** [comment]
Can you all just chill out a bit, no need to get argumentative.

Going to add my two cents to this.

Fighters should be able to be fast, that’s arguably the point of a fighter. 

Brick systems are intentionally supposed to be strong so that fighters can be small and still functional. I haven’t tested the heat stuff on a fighter yet myself but I know that on a 30m ship made from bricks they produce a lot of heat and take a little tweaking to get right, but they do work in the end. 

Not all ships will be created equal and that’s fine.

Finally I don’t think balance is massively important at the moment. We still have a lot of combat related tech to be added and there isn’t a lot of ways to fight at the moment. So while it’s important to fix anything that’s broken, like the shield recharge time, it’s not massively important to get all the finer details ironed out just yet.

**Comment by Drillz007** [comment]
this discussion should just get a pin stuck in it for a good while there is not enough tested information present and continuing this will get us no where with a steam release around the corner there are far more important matters to attend to
